httomo/transform_loader_params.py

```diff
@@ -99,6 +116,54 @@ def conv_param(par: Optional[PreviewParamEntry], length: int):
     )
 
 
+def _keywords_converter(val: Union[int, None, str], length: int, key_type: str) -> int:
+    """
+    Takes the value to assign an index to it. Looks into keywords (begin, mid, end) if they are used.
+    """
+    if isinstance(val, str):
+        if val == "begin":
+            return 0
+        elif val == "mid":
+            start, stop = _get_middle_slice_indices(length)
+            return int(stop + start) // 2
+        elif val == "end":
+            return length
+        else:
+            raise ValueError(
+                f"The given keyword: {val} is not recognised. The recognised keywords are: begin, mid, end."
+            )
```

Comment:
make sense, will do

Reply:
Cool, I saw the added test in 2cef759, thanks!

Maybe I should have been more specific though: I was meaning to test `parse_preview()` with the different keywords, not test `_keywords_converter()` directly. This is because `_keywords_converter()` is marked as private and thus isn't intended to be used standalone. So I don't think the error-raising behavior should be tested by running the private function directly, but by running the public function that calls it, namely, `parse_preview()`.

So instead, I'd recommend adapting the test that was added to work with the public `parse_preview()`.

Reply:
thanks, done.